src/Network/Receive.pm

```diff
@@ -2406,4 +2406,48 @@ sub hat_effect {
 	}
 }
 
+##
+# 096D <size>.W { <index>.W }*
+# @author [Cydh]
+##
+sub merge_item_open {
+	my ($self, $args) = @_;
+	my $num = ($args->{length} - 4) / 2;
+	message TF("Received %d items to be merged. Use 'merge' to continue\n", $num), "info";
+}
+
+sub parse_merge_item_open {
+	my ($self, $args) = @_;
+	@mergeItemList = map {{ itemIndex => unpack('v', $_) }} unpack '(a2)*', $args->{itemList};
```

Comment:
The original idea was for `parse_*` functions to just further unpack `$args` and do nothing except that (do not modify the global state).

For example, `$args->{items} = [map ... $args->{itemList}]` in `parse_merge_item_open` and then assign `@mergeItemList = @{$args->{items}}` in actual `merge_item_open` handler.